Sources/Sentry/SentryTransportAdapter.m

```diff
@@ -94,6 +95,7 @@ - (void)sendUserFeedback:(SentryUserFeedback *)userFeedback
 
 - (void)sendEnvelope:(SentryEnvelope *)envelope
 {
+    envelope.header.sentAt = SentryCurrentDate.date;
```

Comment:
`h`: This sets the sentAt timestamp to when we are caching to disk, which we must not do according to the [docs](https://develop.sentry.dev/sdk/envelopes/):

> For example, SDKs that implement caching features should avoid writing the sent_at header when caching to disk. Only write it when actually sending the event to Sentry.


Instead, we have to set `sentAt` directly before we send it:
https://github.com/getsentry/sentry-cocoa/blob/2cf1b84604c9c8e3a16b9707ef825e4d94abfb48/Sources/Sentry/SentryHttpTransport.m#L281-L283